clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -822,19 +822,25 @@ public void onSuccess(OffsetForEpochResult offsetsResult) {
                         FetchPosition requestPosition = fetchPositions.get(respTopicPartition);
 
                         if (respEndOffset.hasUndefinedEpochOrOffset()) {
-                            handleOffsetOutOfRange(requestPosition, respTopicPartition,
-                                "Failed leader offset epoch validation for " + respEndOffset
-                                + " since no end offset larger than current fetch epoch was reported");
+                            try {
+                                handleOffsetOutOfRange(requestPosition, respTopicPartition,
+                                    "Failed leader offset epoch validation for " + requestPosition
+                                        + " since no end offset larger than current fetch epoch was reported");
+                            } catch (OffsetOutOfRangeException e) {
+                                // Swallow the OffsetOutOfRangeException to finish all partitions validation.
```

Comment:
I don't feel great about having this in the code, even if it's supposed to be temporary. I think we should just fix the exception propagation bug in this patch. It seems like it would be straightforward to do something similar to what is done in the `onFailure` path.
```java
                    if (!(e instanceof RetriableException) && !cachedOffsetForLeaderException.compareAndSet(null, e)) {
                        log.error("Discarding error in OffsetsForLeaderEpoch because another error is pending", e);
                    }
```
Above may not be ideal, but at least it provides a way to propagate individual errors.